session/auth/authprovider/tokenseed.go

```diff
@@ -37,7 +37,7 @@ func (ts *tokenSeeds) getSeed(host string) ([]byte, error) {
 
 	l := flock.New(filepath.Join(ts.dir, ".token_seed.lock"))
 	if err := l.Lock(); err != nil {
-		if !errors.Is(err, syscall.EROFS) && errors.Is(err, syscall.EPERM) {
+		if !errors.Is(err, syscall.EROFS) && !errors.Is(err, os.ErrPermission) {
```

Comment:
@rittneje Was this missing `!` the actual issue?

Reply:
While this was a bug, it wasn't the issue I was seeing. That error was from opening the token seed file (not the lock file), presumably as part of `ioutil.WriteFile`.
https://github.com/moby/buildkit/blob/64dd6d55209057e336d34cf55d7c4b5a06d6ae56/session/auth/authprovider/tokenseed.go#L71-L75

Of note, the nuances between `EPERM` and `EACCES` are confusing, but from what I understand in this circumstance it would give `EACCES`, hence the issue. https://man7.org/linux/man-pages/man2/open.2.html

> EACCES The requested access to the file is not allowed, or search
              permission is denied for one of the directories in the
              path prefix of pathname, or the file did not exist yet and
              write access to the parent directory is not allowed.